source/Calamari/Kubernetes/ResourceStatus/KubernetesYaml.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using System.IO;
+using Calamari.Kubernetes.ResourceStatus.Resources;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+
+namespace Calamari.Kubernetes.ResourceStatus
+{
+    public static class KubernetesYaml
+    {
+        private static readonly IDeserializer Deserializer = new DeserializerBuilder().Build();
+    
+        /// <summary>
+        /// Gets resource identifiers which are defined in a YAML file.
+        /// A YAML file can define multiple resources, separated by '---'.
+        /// </summary>
+        public static IEnumerable<ResourceIdentifier> GetDefinedResources(string manifests)
+        {
+            var input = new StringReader(manifests);
+    
+            var parser = new Parser(input);
+            parser.Consume<StreamStart>();
+    
+            var result = new List<ResourceIdentifier>();
+            while (!parser.Accept<StreamEnd>(out _))
+            {
+                result.Add(GetDefinedResource(parser));
+            }
+    
+            return result;
+        }
+    
+        private static ResourceIdentifier GetDefinedResource(IParser parser)
+        {
+            var yamlObject = Deserializer.Deserialize<dynamic>(parser);
+            yamlObject["metadata"].TryGetValue("namespace", out object @namespace);
```

Comment:
Thanks for capturing this. I've added the logic to skip the status check wrapper if the wrapper that runs `kubectl apply` returns non-zero exit code. So, if we could reach this piece of code, it means `kubectl apply` was successful and so the `metadata` section shouldn't be missing, because it should contain the required information "name". But just to keep it robust, I've also added try-catch around this piece of code so now it omits bad yaml definitions rather than blowing up.